core/src/blocktree.rs

```diff
@@ -980,6 +1134,77 @@ impl Blocktree {
         Ok((blobs, num))
     }
 
+    pub fn get_slot_entries_with_shred_count(
+        &self,
+        slot: u64,
+        start_index: u64,
+    ) -> Result<(Vec<Entry>, usize)> {
+        // Find the next consecutive block of blobs.
+        let serialized_shreds = get_slot_consecutive_shreds(slot, &self.db, start_index)?;
+        trace!(
+            "Found {:?} shreds for slot {:?}",
+            serialized_shreds.len(),
+            slot
+        );
+        let mut shreds: Vec<Shred> = serialized_shreds
+            .iter()
+            .map(|serialzied_shred| {
+                let shred: Shred =
+                    bincode::deserialize(serialzied_shred).expect("Failed to deserialize shred");
+                shred
+            })
+            .collect();
+
+        let mut all_entries = vec![];
+        let mut num = 0;
+        loop {
+            let mut look_for_last_shred = true;
+
+            let mut shred_chunk = vec![];
+            while look_for_last_shred && shreds.len() != 0 {
+                let shred = shreds.remove(0);
+                if let Shred::LastInFECSet(_) = shred {
+                    look_for_last_shred = false;
+                } else if let Shred::LastInSlot(_) = shred {
+                    look_for_last_shred = false;
+                }
+                shred_chunk.push(shred);
+            }
+
+            debug!(
+                "{:?} shreds in last FEC set. Looking for last shred {:?}",
+                shred_chunk.len(),
+                look_for_last_shred
+            );
+
+            // Break if we didn't find the last shred (as more data is required)
+            if look_for_last_shred {
+                break;
+            }
+
+            if let Ok(deshred) = Shredder::deshred(&shred_chunk) {
```

Comment:
sure would be nice for deshred to return a cursor (implements Read) that could be passed to bincode::deserialize_from()...